src/Zipper.elm

```diff
@@ -1014,36 +946,39 @@ changeToAlpha z =
             (\tableau ->
                 -- pozor na koren
                 case tableau.ext of
+                    Alpha t ->
+                        Tableau tableau.node (extType t)
+
                     Beta lt rt ->
                         if lt.node.value == "" then
-                            Tableau tableau.node (Alpha rt)
+                            Tableau tableau.node (extType rt)
 
                         else if rt.node.value == "" then
-                            Tableau tableau.node (Alpha lt)
+                            Tableau tableau.node (extType lt)
 
                         else
                             Tableau tableau.node (Beta lt rt)
 
                     Gamma t s ->
-                        Tableau tableau.node (Alpha t)
+                        Tableau tableau.node (extType t)
 
                     Delta t s ->
-                        Tableau tableau.node (Alpha t)
+                        Tableau tableau.node (extType t)
 
                     Refl t ->
-                        Tableau tableau.node (Alpha t)
+                        Tableau tableau.node (extType t)
 
                     Leibnitz t ->
-                        Tableau tableau.node (Alpha t)
+                        Tableau tableau.node (extType t)
 
                     _ ->
                         tableau
             )
             (z |> up)
```

Comment:
Basically, the same thing is done for all alternatives except for `Beta`. How about defining:
```elm
leftSubtree : Extension -> Tableau
```
which will return the only sub-tableau of unary extensions and the left subtableau of `Beta`?
You could then handle all alternatives except `Beta` in the default (`_`).

You could also generalize this completely if you defined also:
```elm
rightSubtree : Extension -> Tableau
```
which would return the actual right subtableau for `Beta` and `Tableau defNode Open` for everything else.

Then you could have a single function:
```elm
changeRule : (Tableau -> Tableau -> Extension) -> Zipper -> Zipper
```
which would call the new extension function with the results of `leftSubtree` and `rightSubtree`.

`changeToUnaryRule`, `changeToUnaryRuleWithSubst`, and `changeToBeta` can all be derived as special cases of this `changeRule`.